core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java

```diff
@@ -988,18 +1031,7 @@ protected boolean isMultipartRequest(HttpServletRequest request) {
      * @return a multi part request object
      */
     protected MultiPartRequest getMultiPartRequest() {
-        MultiPartRequest mpr = null;
-        //check for alternate implementations of MultiPartRequest
-        Set<String> multiNames = getContainer().getInstanceNames(MultiPartRequest.class);
-        for (String multiName : multiNames) {
-            if (multiName.equals(multipartHandlerName)) {
-                mpr = getContainer().getInstance(MultiPartRequest.class, multiName);
-            }
-        }
-        if (mpr == null) {
-            mpr = getContainer().getInstance(MultiPartRequest.class);
-        }
-        return mpr;
+        return getContainer().getInstance(MultiPartRequest.class);
```

Comment:
Shouldn't it be an instance field as other?

Reply:
`MultiPartRequest` is prototype scope and we need a unique instance per request so injecting it once wouldn't work here